pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/PbjMap.java

```diff
@@ -0,0 +1,137 @@
+package com.hedera.pbj.runtime;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Implements an immutable map that exposes a list of keys sorted in their natural order.
+ * <p>
+ * This Map implementation allows one to iterate the entries in a deterministic order
+ * which is useful for serializing, hash computation, etc.
+ *
+ * @param <K> key type
+ * @param <V> value type
+ */
+public class PbjMap<K, V> implements Map<K, V> {
```

Comment:
What is the advantage of this over `Collections.unmodifiableSortedMap()` ?

Reply:
Performance and code simplicity.

The `Collections.unmodifiableSortedMap()` accepts a `SortedMap` already (it cannot take a regular `Map`), so for starters we'd have to build one when we're parsing a message. Building a SortedMap (e.g. a TreeMap) that keeps its whole entries in a particular order may be a bit less efficient than building a simple HashMap and then only sorting a set of its keys.

Further, regular map operations (e.g. `get(key)`) would be O(log N) for TreeMap instead of O(1) for HashMap, further degrading the performance of the client code that uses the parsed model object and queries the map.

So I went with the new `PbjMap` instead.